sui/api/tests/test_graphql.rs

```diff
@@ -0,0 +1,162 @@
+use aux_rs::client::{AuxClient, CoinClient, PoolClient};
+use aux_rs::schema::{Coin, Coins, Curve, FeeTier, Percent, PoolInput};
+use color_eyre::eyre::{eyre, Result};
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+use serde::Deserialize;
+use serde_json::json;
+use std::path::Path;
+use sui_framework_build::compiled_package::BuildConfig;
+use sui_keys::keystore::{AccountKeystore, FileBasedKeystore, Keystore};
+use sui_sdk::{types::base_types::ObjectID, SuiClient};
+use sui_types::messages::{ExecuteTransactionRequestType, Transaction};
+
+#[derive(Debug, Deserialize)]
+struct GraphQLResponse<T> {
+    pub data: Option<T>,
+    pub errors: Option<serde_json::Value>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Pool {
+    pub id: String,
+    pub type_: String,
+    pub coins: Vec<Coin>,
+    pub amounts: Vec<u64>,
+}
+
+async fn publish(sui_client: &SuiClient, keystore: &Keystore) -> ObjectID {
+    let compiled_modules = BuildConfig::default()
+        .build(Path::new("../contract").to_path_buf())
+        .unwrap()
+        .get_package_bytes();
+    let sender = *keystore.addresses().last().unwrap();
+    let tx = sui_client
+        .transaction_builder()
+        .publish(sender, compiled_modules, None, 30_000)
+        .await
+        .unwrap();
+
+    let signer = *keystore.addresses().last().unwrap();
+    let signature = keystore.sign(&signer, &tx.to_bytes()).unwrap();
+    let tx = sui_client
+        .quorum_driver()
+        .execute_transaction(
+            Transaction::from_data(tx, signature).verify().unwrap(),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+        )
+        .await
+        .unwrap();
+    *tx.effects
+        .unwrap()
+        .events
+        .iter()
+        .filter_map(|event| match event {
+            sui_sdk::rpc_types::SuiEvent::MoveEvent {
+                package_id, type_, ..
+            } if type_.ends_with("PoolsCreated") => Some(package_id),
+            _ => None,
+        })
+        .next()
+        .unwrap()
+}
+
+#[tokio::test]
+async fn test_graphql() -> Result<()> {
```

Comment:
@leina05 sorry for the large PR, recommend this test as a starting point for how things wire together